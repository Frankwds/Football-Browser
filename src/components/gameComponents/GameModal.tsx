import Rating from "./Rating";

import Comments from "./Comments";
import { ModalBody } from "@chakra-ui/react";
import { useQuery, gql } from "@apollo/client";

type Props = {
  id: string;
  country: string;
  series: string;
  homeTeam: string;
  awayTeam: string;
  callbackOnRate: (callbackData: any) => void;
  callbackOnComment: (comments: string[]) => void;
  rating: number;
  numReviews: number;
  comments: string[];
};

const GETSINGLEGAME = gql`
  query GetGameDetailsByID($gameID: String!) {
    GetGameByID(gameID: $gameID) {
      date
      season
      fthg
      ftag
    }
  }
`;

/**
 * GameModal - Modal presenting information about football match
 *
 * @param props
 *
 * @returns {React.FC}
 *
 * Gives detailed information about football match.
 * Data is retrieved by means of the GraphQL query @constant GETSINGLEGAME through
 * the @function useQuery in the @link @apollo/client library.
 */

const GameModal: React.FC<Props> = (props) => {
  const { loading, error, data } = useQuery(GETSINGLEGAME, {
    variables: {
      gameID: props.id,
    },
  });

  //Presents game score of match
  const score = () => {
    let wt = data.GetGameByID.fthg;
    let wtn = props.homeTeam;
    let lt = data.GetGameByID.ftag;

    if (lt > wt) {
      wt = lt;
      lt = data.GetGameByID.fthg;
      wtn = props.awayTeam;
    } else if (lt == wt) {
      return (
        <p>
          Final score: {wt} - {lt}
        </p>
      );
    }
    return (
      <p>
        Final score: {wt} - {lt} to {wtn}!
      </p>
    );
  };
  if (loading) {
    return <div>Loading...</div>;
  }
  if (error) {
    return <div>404: Could not find data!</div>;
  }

  return (
    <ModalBody>
      <p>The id of the game is: {props.id}</p>
      <ul>
        <li>League: {props.series}</li>
        <li>Country: {props.country} </li>
        <li>Season: {data.GetGameByID.season}</li>
        <li>Date: {data.GetGameByID.date}</li>
        <li>Home team: {props.homeTeam}</li>
        <li>Away team: {props.awayTeam}</li>
        <li>{score()}</li>
      </ul>
      <br />
      <Rating
        id={props.id}
        rating={props.rating}
        numReviews={props.numReviews}
        callbackOnRate={props.callbackOnRate}
      ></Rating>
      <br />
      <Comments
        id={props.id}
        comments={props.comments}
        callbackFunc={props.callbackOnComment}
      />
      <br />
    </ModalBody>
  );
};

export default GameModal;
