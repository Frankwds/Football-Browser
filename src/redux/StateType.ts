import {SearchData} from "./searchFilter/searchFilterTypes"; 

//A type interface for the only implemented query, scalable with more reducers.
export interface StateType{
    searchQueries: SearchData,
}


