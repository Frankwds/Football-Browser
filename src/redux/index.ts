import { combineReducers } from "redux";
import {searchReducer} from "./searchFilter/filterSearchReducer";


// In this case only one reducer, but very scalable.
const allReducers = combineReducers({
  searchQueries: searchReducer,
});

export default allReducers;