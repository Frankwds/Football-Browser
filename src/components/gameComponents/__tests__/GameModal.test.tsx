import { ChakraProvider } from "@chakra-ui/react";
import { useState } from "react";
import ReactDOM from "react-dom";
import { act } from "react-dom/test-utils";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { createStore } from "redux";
import allReducers from "../../../redux";
import GameModal from "../GameModal";

let container: any;
const store = createStore(allReducers);

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe("Testing GameModal", () => {
  it.skip("should render without crashing", () => {
    act(() => {
      const div = document.createElement("div");
      ReactDOM.render(
        <Provider store={store}>
          <ChakraProvider>
            <GameModal
              id={"testId"}
              country={"Norway"}
              series={"N1"}
              homeTeam={"GGTeam"}
              awayTeam={"FrankTeam"}
              callbackOnRate={() => {}}
              callbackOnComment={() => {}}
              rating={3}
              numReviews={666}
              comments={["1", "2", "3"]}
            />
          </ChakraProvider>
        </Provider>,
        container
      );
    });
  });

  it.skip("snapshot should be same as previous", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ChakraProvider>
            <GameModal
              id={"testId"}
              country={"Norway"}
              series={"N1"}
              homeTeam={"GGTeam"}
              awayTeam={"FrankTeam"}
              callbackOnRate={() => {}}
              callbackOnComment={() => {}}
              rating={3}
              numReviews={666}
              comments={["1", "2", "3"]}
            />
          </ChakraProvider>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
