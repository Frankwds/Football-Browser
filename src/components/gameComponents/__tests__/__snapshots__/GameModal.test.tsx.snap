// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GameModal snapshot should be same as previous 1`] = `
Array [
  <div>
    Loading...
  </div>,
  <span />,
]
`;
